setup/advancedConfig.yaml

```diff
@@ -150,6 +150,7 @@ azstorage:
   cpk-enabled: true|false <enable client provided key encryption>
   cpk-encryption-key: <customer provided base64-encoded AES-256 encryption key value>
   cpk-encryption-key-sha256:  <customer provided base64-encoded sha256 of the encryption key>
+  preserve-acl: true|false <preserve ACLs and Permissions set on file during updates>
```

Comment:
Shouldn't we make it default behaviour, rather than rely over a flag.
We can discuss with Vishnu once, but In my opinion, it should be default behaviour, rather than relying over a flag.

Reply:
By default we are not doing this in existing builds. If we enable by default then it means 2 additional REST calls for all customers hitting this flow which will degrade performance and cost mroe to customers. This shall hit only if customers really want to preserve the permissions. Ideally it shall always do but just to avoid extra cost we need the flag.

Reply:
Correctness of the tool should be the priority imo.

Reply:
If we make this the default behavior, it will break many customers workflows who use OAuth with `Storage Blob Data Contributor` role. This is the RBAC role generally being used and it fails on SetAccessControl API. This API only works with `Storage Blob Data Owner` role.
Also, if some customers only care about perf and not ACLs, then making this the default behavior will also impact them.

Reply:
SetAccessControl is also used while creating the file. 

now setting them when creating the file and droping them again while updating the file make sense?

Reply:
Creation time we do it because user sends the permissions while creating a file and we need to save those to file. This path is for any upload/update. So for creation of a new file I do see that there is an overhead.

Reply:
Not able to come up with any way to skip this on upload as we never know during upload whether the file was created newly or its updating of an existing file.